### ISSUE TITLE
Is it possible to pass parameter to Relationship Function?

### ISSUE THREAD

**Issue by likemusic** [analogueorm/analogue#39]
Is it possible to pass parameter to Relationship Function?

Something like there:

``` php
public function achievements($orderBy = true)
{
$result = $this->belongsToMany (...)
if($orderBy) return $result->orderBy(...)
return $result;
}

$member->achievements(true)->get()->(...)
```

(code from http://stackoverflow.com/questions/23016216/laravel-pass-parameter-to-relationship-function )

**Comment by likemusic** [comment]
Currently it's impossible. I'm implement this in local forked repository, but now I don't have time for rewrite and run tests.

**Comment by likemusic** [comment]
Not tested Impementation here - https://github.com/likemusic/analogue/tree/relationship-with-params

More changes here:
 - Delete relationship multiple instanting
   - On instanting Model, for classification relation type (one or collection)
   - On load results from proxy
 - Delete using EntityProxy (no needs as i see)

**Comment by siempay** [comment]
use Illuminate\Foundation\Auth\User as Authenticatable;

class User extends Authenticatable
{
    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'name', 'email', 'password',
    ];

    /**
     * The attributes that should be hidden for arrays.
     *
     * @var array
     */
    protected $hidden = [
        'password', 'remember_token',
    ];

 public function panier(status_id){
        return $this->hasMany('App\panier')->where("status_id",status_id);
    }


I wanna do somthing like this? is it possible?

**Comment by RemiCollin** [comment]
Implemented in upcoming 5.5 release

**Comment by sobhanattar** [comment]
@RemiCollin can you please point to the change?
Thanks
